### PR TITLE
net: ieee802154: Add new config options

### PR DESCRIPTION
**Pull request by rlubos**
Add `IEEE802154_CONFIG_PROMISCUOUS` and `IEEE802154_CONFIG_PAN_COORDINATOR` config options.

### CODE REVIEW

File: drivers/ieee802154/ieee802154_nrf5.c [hunk 1]
```diff
@@ -490,6 +490,10 @@ int nrf5_configure(struct device *dev, enum ieee802154_config_type type,
 
 		break;
 
+	case IEEE802154_CONFIG_PAN_COORDINATOR:
+		nrf_802154_pan_coord_set(config->pan_coordinator);
```
Comment (tbursztyka) [inline drivers/ieee802154/ieee802154_nrf5.c]: too bad I first thought you would be implementing PAN support in the l2 ;)

**Review by tbursztyka — APPROVED** [review]
sounds good to me

**Review by jukkar — APPROVED** [review]

**Comment by markus-becker-tridonic-com** [comment]
@rlubos, would this allow for building an NCP radio sniffer with https://openthread.io/guides/pyspinel/sniffer?

**Comment by rlubos** [comment]
@markus-becker-tridonic-com I haven' tested the sniffer solution so i can' tell for sure.
Promiscuous mode is certainly needed for a sniffer, so if it was the only missing piece then indeed, there's a chance it would work (not I've only added implementation of this config for nRF).

**Pull request merged**